server/forge/bitbucketdatacenter/bitbucketdatacenter.go

```diff
@@ -364,22 +364,28 @@ func (c *client) BranchHead(ctx context.Context, u *model.User, r *model.Repo, b
 	if err != nil {
 		return nil, fmt.Errorf("unable to create bitbucket client: %w", err)
 	}
-	branches, _, err := bc.Projects.SearchBranches(ctx, r.Owner, r.Name, &bb.BranchSearchOptions{Filter: b})
+	commits, _, err := bc.Projects.SearchCommits(ctx, r.Owner, r.Name, &bb.CommitSearchOptions{
```

Comment:
I'll see if I can do this API call to Bitbucket to verify that the `Until` property actually returns what we expect. The `b` variable will contain the branch name not the ref?